lib/gs-plugin-loader.c

```diff
@@ -2262,6 +2263,7 @@ gs_plugin_loader_status_changed_cb (GsPlugin *plugin,
 		}
 		return;
 	}
+	g_hash_table_insert(priv->global_plugin_statuses, gs_app_get_id(app), (guint)status);
```

Comment:
note that you shouldn't use integers as values in a `GHashTable` since all values are treated as pointers.

You need to use the `GUINT_TO_POINTER()` and `GPOINTER_TO_UINT()` macros when inserting and retrieving integer values.

Here's a good explanation (from a former Endlesser!): https://stackoverflow.com/a/7124255

Reference docs here: https://developer.gnome.org/glib/stable/glib-Type-Conversion-Macros.html

Reply:
Thanks for the reference. Made the change.